packages/entity/src/internal/ReadThroughEntityCache.ts

```diff
@@ -32,113 +33,122 @@ export default class ReadThroughEntityCache<TFields extends Record<string, any>>
+      fetcherValues: readonly TLoadValue[],
+    ) => Promise<ReadonlyMap<TLoadValue, readonly Readonly<TFields>[]>>,
+  ): Promise<ReadonlyMap<TLoadValue, readonly Readonly<TFields>[]>> {
+    // return normal fetch when cache by key not supported
+    if (!this.isLoadKeyCacheable(key)) {
+      return await fetcher(values);
     }
 
-    const cacheLoadResults = await this.entityCacheAdapter.loadManyAsync(fieldName, fieldValues);
+    const cacheLoadResults = await this.entityCacheAdapter.loadManyAsync(key, values);
 
     invariant(
-      cacheLoadResults.size === fieldValues.length,
-      `${this.constructor.name} loadMany should return a result for each fieldValue`,
+      cacheLoadResults.size === values.length,
+      `${this.constructor.name} loadMany should return a result for each value`,
     );
 
-    const fieldValuesToFetchFromDB = Array.from(
+    const valuesToFetchFromDB = Array.from(
       filterMap(
         cacheLoadResults,
         (cacheLoadResult) => cacheLoadResult.status === CacheStatus.MISS,
       ).keys(),
     );
 
     // put transformed cache hits in result map
-    const results: Map<NonNullable<TFields[N]>, readonly Readonly<TFields>[]> = new Map();
-    cacheLoadResults.forEach((cacheLoadResult, fieldValue) => {
+    const results = key.vendNewLoadValueMap<readonly Readonly<TFields>[]>();
+    cacheLoadResults.forEach((cacheLoadResult, value) => {
       if (cacheLoadResult.status === CacheStatus.HIT) {
-        results.set(fieldValue, [cacheLoadResult.item]);
+        results.set(value, [cacheLoadResult.item]);
       }
     });
 
     // fetch any misses from DB, add DB objects to results, cache DB results, inform cache of any missing DB results
-    if (fieldValuesToFetchFromDB.length > 0) {
-      const dbFetchResults = await fetcher(fieldValuesToFetchFromDB);
+    if (valuesToFetchFromDB.length > 0) {
+      const dbFetchResults = await fetcher(valuesToFetchFromDB);
 
-      const fieldValueDBMisses = fieldValuesToFetchFromDB.filter((fv) => {
+      const valueDBMisses = valuesToFetchFromDB.filter((fv) => {
         const objectsFromFulfillerForFv = dbFetchResults.get(fv);
         return !objectsFromFulfillerForFv || objectsFromFulfillerForFv.length === 0;
       });
 
-      const objectsToCache: Map<NonNullable<TFields[N]>, Readonly<TFields>> = new Map();
-      for (const [fieldValue, objects] of dbFetchResults.entries()) {
+      const objectsToCache = key.vendNewLoadValueMap<Readonly<TFields>>();
+      for (const [value, objects] of dbFetchResults.entries()) {
         if (objects.length > 1) {
           // multiple objects received for what was supposed to be a unique query, don't add to return map nor cache
           // TODO(wschurman): emit or throw here since console may not be available
           // eslint-disable-next-line no-console
           console.warn(
-            `unique key ${String(fieldName)} in ${
+            `unique key ${key.debugString()} in table ${
```

Comment:
Could override toString() instead. Up to you.

Reply:
Ended up doing this in the top PR.